AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/AWSAuthSignInPluginTests.swift

```diff
@@ -585,6 +585,56 @@ class AWSAuthSignInPluginTests: BasePluginTest {
         }
     }
 
+    /// Test a signIn with customAuthWIthoutSRP
+    ///
+    /// - Given: Given an auth plugin with mocked service. Returning a new challenge after confirm sign in is called
+    ///
+    /// - When:
+    ///    - I invoke signIn and then confirm sign in
+    /// - Then:
+    ///    - The next step smsMfA should be triggered
+    ///
+    func testSignInWithCustomAuthIncorrectCode() async {
+
+        self.mockIdentityProvider = MockIdentityProvider(mockInitiateAuthResponse: { _ in
+            InitiateAuthOutputResponse(
+                authenticationResult: .none,
+                challengeName: .customChallenge,
+                challengeParameters: InitiateAuthOutputResponse.validChalengeParams,
+                session: "someSession")
+        }, mockRespondToAuthChallengeResponse: { _ in
+            RespondToAuthChallengeOutputResponse(
+                authenticationResult: .none,
+                challengeName: .smsMfa,
+                challengeParameters: ["paramKey": "value"],
+                session: "session")
+        })
+
+        let pluginOptions = AWSAuthSignInOptions(validationData: ["somekey": "somevalue"],
+                                                 metadata: ["somekey": "somevalue"],
+                                                 authFlowType: .customWithoutSRP)
+        let options = AuthSignInRequest.Options(pluginOptions: pluginOptions)
+        do {
+            let result = try await plugin.signIn(
+                username: "username",
+                password: "password",
+                options: options)
+            if case .confirmSignInWithCustomChallenge = result.nextStep {
+                let confirmSignInResult = try await plugin.confirmSignIn(challengeResponse: "245234")
+                if case .confirmSignInWithSMSMFACode = confirmSignInResult.nextStep {
+
+                } else {
+                    XCTFail("Incorrect challenge type")
+                }
+            } else {
+                XCTFail("Incorrect challenge type")
+            }
```

Comment:
Recommend chaining a single guard to represent the expected path.
```suggestion
            guard case .confirmSignInWithCustomChallenge = result.nextStep,
                  case let confirmSignInResult = try await plugin.confirmSignIn(
                    challengeResponse: "245234"
                  ),
                  case .confirmSignInWithSMSMFACode = confirmSignInResult.nextStep
            else {
                return XCTFail("Incorrect challenge type")
            }
```